runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java

```diff
@@ -75,10 +75,15 @@
 	private static final Pattern BINDING_USED_PATTERN = Pattern.compile(
 		"You bind the temple's power into (mud|lava|steam|dust|smoke|mist) runes\\.");
 	private static final String BINDING_BREAK_TEXT = "Your Binding necklace has disintegrated.";
+	private static final Pattern RING_OF_FORGING_CHECK_PATTERN = Pattern.compile(
+		"You can smelt ([0-9+]+|one) more pieces of iron ore before a ring melts\\.");
+	private static final String RING_OF_FORGING_USED_TEXT = "You retrieve a bar of iron.";
+	private static final String RING_OF_FORGING_BREAK_TEXT = "Your Ring of Forging has melted.";
```

Comment:
This message has some color applied, so this is not the correct exact string.

![Ring of forging melt text](https://cdn.discordapp.com/attachments/419891709883973642/601102203297726465/unknown.png)

Reply:
Oh alright! I wasn't aware that colour was coded into the string itself. That explains why the recoil text looks odd too. Do you recommend using the same syntax and if so, which value is the purple text?

Reply:
I'd recommend using the same syntax as the recoil text. Sadly, I don't know the value (not my screenshot), but the color from the image is `rgb(127, 0, 127)`, which would be `"<col=7f007f>"`.

Reply:
Ok thanks! I just checked in game and the next Ring of Forging has 140 charges so the pattern matcher must have caught the message. I will alter the code anyway for the sake of correctness.